apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -707,6 +801,27 @@ static void tc_fs_vfs_rewinddir(void)
 	ret = closedir(dirp);
 	TC_ASSERT_EQ("closedir", ret, OK);
 	TC_ASSERT_EQ("rewinddir", count, VFS_LOOP_COUNT * 2);
+
+	/*For Pseudo dir operations */
+	dirp = opendir(ROOT);
+	TC_ASSERT_NEQ("opendir", dirp, NULL);
+
+	count = 0;
+	temp_count = 0;
+
+	while ((dirent = readdir(dirp)) != NULL) {
+		count++;
+	}
+
+	rewinddir(dirp);
+
+	while ((dirent = readdir(dirp)) != NULL) {
+		temp_count++;
+	}
```

Comment:
I think, we can save an int(temp_count) on stack. :)
Like,
for (count = 0; readdir(dirp); count++);                                            
for (rewinddir(dirp); readdir(dirp); count--);     
Then,                                 
TC_ASSERT_EQ("rewinddir", count, 0);

Will it work , i'm just curious?  :)

Reply:
Okay will check.

Reply:
No, in 2nd loop will run infinetly because smartfs_rewinddir always return OK

Reply:
The above suggested code is working fine.

Reply:
Even if it works fine, it is not good choice.
Using function which can changed value dynamically, it has potential defect.
Actually some defect checking tool(i.e coverity) treats this kind of code as a defect, as I experienced.
Of course it is safe because this is tc but If another application delete folder in those folder, it will fail.
That's why defect tools usually treats it as a defect.